runtime/interpreter/value_string.go

```diff
@@ -84,22 +84,14 @@ func stringFunctionFromCharacters(invocation Invocation) Value {
 
 	inter := invocation.Interpreter
 
-	common.UseMemory(inter,
-		common.MemoryUsage{
-			Kind:   common.MemoryKindStringValue,
-			Amount: 1,
-		},
-	)
+	// NewStringMemoryUsage already accounts for empty string.
+	common.UseMemory(inter, common.NewStringMemoryUsage(0))
 	var builder strings.Builder
 
 	argument.Iterate(inter, func(element Value) (resume bool) {
 		character := element.(CharacterValue)
-		common.UseMemory(inter,
-			common.MemoryUsage{
-				Kind:   common.MemoryKindStringValue,
-				Amount: uint64(len(character)),
-			},
-		)
+		// -1 to offset the double counting for empty string.
+		common.UseMemory(inter, common.NewStringMemoryUsage(len(character)-1))
```

Comment:
It looks a bit odd to decrement here (though I understand the reason). Maybe that's probably why the original code did it without the constructor. Maybe keep it in the original way, but adding a comment explaining why it is done this way might be better?
cc: @turbolent

Reply:
Personally, I prefer how it was written before. Processing why the extra `-1` is used could take a few seconds and can be avoided in this case. I've updated the code and added the clarification comment.

Reply:
Yeah, whatever is clearest is best 👍